src/net.c

```diff
@@ -1836,6 +1845,27 @@ handle_request(coap_context_t *context, coap_session_t *session, coap_pdu_t *pdu
 	response = coap_new_error_response(pdu, COAP_RESPONSE_CODE(405),
 	  opt_filter);
       }
+    } else if (context->unknown_resource) {
+      /*
+       * The unknown_resource can be used to handle undefined resources
+       * and can support any registered handler defined for it
+       * Example set up code:-
+       *   r = coap_resource_init((unsigned char *)"unknown_resource", 20, 0);
```

Comment:
I would suggest to use at least a more exotic name for the unknown resource example name to make naming conflicts less likely. 

For example:

```c
r = COAP_UNKNOWN_RESOURCE;
```

where 

```c
const str coap_unknown_uri = { 1, (unsigned char *)":" };
#define COAP_UNKNOWN_RESOURCE coap_resource_init(coap_unknown_uri.s, coap_unknown_uri.length, 0)
```